### ISSUE TITLE
Does not follow theme on theme change

### ISSUE THREAD

**Issue by Bytezz** [lundmar/gtkchart#3]
When changing theme (for example switching light/dark), the widget keeps following the previous one.

[Switching light/dark](https://user-images.githubusercontent.com/25850701/193054880-17a3e47b-2eed-47a7-abc0-9797ae6001b2.webm)

**Comment by lundmar** [comment]
Oh I see. So it is actually switching but it looks inverted.

I'm currently working on adding a demo application so I can reproduce this.

**Comment by lundmar** [comment]
Okay, so now I've added a simple demo application that is easy to build via meson.

This theme switching issue may not be a problem with our implementation. In fact, the theme switching seems to work perfectly fine when I use the new quick settings dark mode toggle option on my Ubuntu 22.10 pre-release installation. However, when using the gnome settings application things does not work quite as well. I suspect this may be a signalling bug in Gnome. Further investigation is required.

[Screencast from 2022-09-29 23-24-01 - gtkchart-debug.webm](https://user-images.githubusercontent.com/353407/193145241-11da0769-11ea-49e4-badc-be45c0cd030d.webm)

**Comment by lundmar** [comment]
It turns out that to capture the theme change signalling correctly the application must be an libadwaita featured application so I will convert the demo application to use libadwaita. Also, we don't actually need to subscribe and react to the theme change signals. A GTK4 developer (Mikhaylenko) kindly hinted that it is easier to simply look up the colors at each snapshot redraw.

I'll make these changes ASAP and test.

**Comment by lundmar** [comment]
@Bytezz Ok, I got everything working now. Updating the colors at each snapshot redraw and converting the demo application to libadwaita make colors update correctly. See video.

This means you will have to convert your baby monitor application to use libadwaita like I did with the demo application or do as explained here: https://gnome.pages.gitlab.gnome.org/libadwaita/doc/1-latest/initialization.html

[Screencast from 2022-09-30 16-06-43 - gtkchart-fixed.webm](https://user-images.githubusercontent.com/353407/193289610-34b59005-1bc7-4f60-ba94-ecf34d601a4b.webm)
